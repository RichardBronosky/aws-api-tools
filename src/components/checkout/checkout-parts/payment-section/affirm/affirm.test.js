import React from 'react'
import renderer from 'react-test-renderer'
import Affirm from './affirm-button'

describe('Affirm', () => {
  const testProps = {
    affirmLoaded: false,
    affirmFunc: jest.fn(),
    affirm_config: {
      public_api_key: 'test-key',
      script: `test-url/js/v2/affirm.js`,
    },
  }

  beforeAll(() => {
    global.window = Object.create(window)
    Object.defineProperty(window, 'affirm', {
      checkout: jest.fn(),
      open: jest.fn(),
    })
  })

  it('renders correctly when affirm is not loaded', () => {
    const tree = renderer.create(<Affirm { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when affirm is loaded', () => {
    const tree = renderer.create(<Affirm { ...{ ...testProps, affirmLoaded: true } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
