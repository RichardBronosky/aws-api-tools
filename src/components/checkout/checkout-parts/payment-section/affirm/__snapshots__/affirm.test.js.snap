// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Affirm renders correctly when affirm is loaded 1`] = `
<button
  className="affirm-button"
  value="Affirm Checkout"
>
  Pay with 
  <img
    alt="Affirm Logo"
    src="test-file-stub"
  />
</button>
`;

exports[`Affirm renders correctly when affirm is not loaded 1`] = `null`;
